File: react-native/react-native-totvs-clockin-vision/android/src/main/java/com/totvs/clockin/vision/events/OnLiveness.kt
package com.totvs.clockin.vision.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReactContext
import com.facebook.react.uimanager.events.RCTEventEmitter

/**
 * Event emitted when liveness feature is enabled and the kind of liveness
 * is triggered.
 */
class OnLiveness(private val mode: Int) : Event {
    override fun invoke(context: ReactContext, viewId: Int) {
        val event = Arguments.createMap().apply {
            putInt(FIELD_EVENT_MODE, mode)
        }
        context.getJSModule(RCTEventEmitter::class.java).receiveEvent(viewId, name, event)
    }

    companion object : Event.Export {
        override val name = "onLiveness"

        private const val FIELD_EVENT_MODE = "mode"
    }
}

File: react-native/react-native-totvs-clockin-vision/ios/Events/OnLiveness.swift
//
//  OnLiveness.swift
//  TOTVSClockInVision
//
//  Created by Jansel Rodriguez on 8/3/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

fileprivate let FIELD_EVENT_MODE = "mode"
/**
* Event emitted when liveness feature is enabled and the kind of liveness
* is triggered.
*/
public class OnLiveness : Event {
    
    private let emit: RCTDirectEventBlock?
    
    public required init(emit: RCTDirectEventBlock?) {
        self.emit = emit
    }
    
    public func send(data: LivenessResult) {
        let event:[String: Any] = [
            FIELD_EVENT_MODE: data.mode
        ]
        emit?(event)
    }
}
